### PR TITLE
Fix email notification link url

### PR DESCRIPTION
**Pull request by lchanmann**
## Ref
- https://nanopay.atlassian.net/browse/NP-3111

## Changes
- Use group.url for notification email link
- Load theme via user spid when the loaded theme's (via domain) spid is not same as user's
- Reload theme when the context theme is available for use in EmailsUtility but the theme's spid is not same as user's

### CODE REVIEW

File: src/foam/nanos/notification/NotificationSetting.js [hunk 1]
```diff
@@ -67,21 +67,8 @@ foam.CLASS({
         { name: 'notification', type: 'foam.nanos.notification.Notification' }
       ],
       javaCode: `
-        X userX = x;
```
Comment (jlhughes) [inline src/foam/nanos/notification/NotificationSetting.js]: @lchanmann why are you removing this? 
I intentionally added it recently, so all downstream - from notificationSettings - logic didn't have to worry about this.

**Review by jlhughes — COMMENTED** [review]
.

File: src/foam/nanos/notification/NotificationSetting.js (hunk 1 above)
Comment (lchanmann) [reply src/foam/nanos/notification/NotificationSetting.js]: Reverted in https://github.com/foam-framework/foam2/pull/4617/commits/816722dda2f442fd04c5ed9425fe5917936c42b5.

File: src/foam/nanos/theme/Themes.js [hunk 2]
```diff
@@ -196,17 +203,6 @@ Later themes:
         }
       }
 
-      if ( theme == null ) {
```
Comment (jlhughes) [inline src/foam/nanos/theme/Themes.js:199]: Why did you remove this?

File: src/foam/nanos/theme/Themes.js [hunk 3]
```diff
@@ -174,9 +175,15 @@ Later themes:
         }
       }
 
+      if ( theme == null ) {
```
Comment (jlhughes) [inline src/foam/nanos/theme/Themes.js:178]: Why did you add this back?

File: src/foam/nanos/theme/Themes.js (hunk 2 above)
Comment (lchanmann) [reply src/foam/nanos/theme/Themes.js:199]: It was moved up to line 178.

File: src/foam/nanos/theme/Themes.js (hunk 3 above)
Comment (lchanmann) [reply src/foam/nanos/theme/Themes.js:178]: It allows loading group theme when there is no spid theme.

**Review by jlhughes — APPROVED** [review]
